core/math/camera_matrix.cpp

```diff
@@ -183,6 +183,10 @@ void CameraMatrix::set_orthogonal(real_t p_size, real_t p_aspect, real_t p_znear
 
 void CameraMatrix::set_frustum(real_t p_left, real_t p_right, real_t p_bottom, real_t p_top, real_t p_near, real_t p_far) {
 
+	ERR_FAIL_COND(Math::is_equal_approx(p_right, p_left));
+	ERR_FAIL_COND(Math::is_equal_approx(p_top, p_bottom));
+	ERR_FAIL_COND(Math::is_equal_approx(p_far, p_near));
```

Comment:
I think we can be more clever here. The code reads like there are expectations on what number should be higher than the other, so we should check that.
```
ERR_FAIL_COND(p_right <= p_left);
ERR_FAIL_COND(p_top <= p_bottom);
ERR_FAIL_COND(p_far <= p_near);
```
CC @JFonS

Reply:
Yes, that makes more sense.